### ISSUE TITLE
fltk_gl_SHARED.lib cannot be opened

### ISSUE THREAD

**Issue by szukw000** [fltk/fltk#433]
```
fltk-version 1.4.0
platform Windows10
Visual Studio 2022 version 17.1.4
cmake version 3.23.1

cmake -DFL_ABI_VERSION:string="10400" -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX:path="C:/Users/Public" -DOPTION_BUILD_SHARED_LIBS=on -DOPTION_USE_THREADS=on -DOPTION_LARGE_FILE=on -DOPTION_USE_SYSTEM_ZLIB=on -DOPTION_USE_SYSTEM_LIBJPEG=on -DOPTION_USE_SYSTEM_LIBPNG=on -DOPTION_USE_XFT=off ..


It seems that _GL does not work with Windows10 AND nmake but exclusively
with MinGW:

===============================================
2022-04-14-error1.txt (last error message):
===============================================


unittest_simple_terminal.cxx
[ 99%] Linking CXX executable ..\bin\test\unittests-shared.exe
LINK: command "C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx86\x64\link.exe /nologo @CMakeFiles\unittests-shared.dir\objects1.rsp /out:..\bin\test\unittests-shared.exe /implib:..\bin\test\unittests-shared.lib /pdb:C:\Users\szukw000\LIB\FLTK\V1.4\fltk-1.4.x-2022-04-14-1\BUILD\bin\test\unittests-shared.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:windows fltk_gl_SHARED.lib ..\lib\libfltk.lib opengl32.lib glu32.lib comctl32.lib ws2_32.lib C:\Users\Public\lib\jpeg.lib C:\Users\Public\lib\png.lib C:\Users\Public\lib\z.lib opengl32.lib glu32.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:..\bin\test\unittests-shared.exe.manifest" failed (exit code 1181) with the following output:
LINK : fatal error LNK1181: Eingabedatei "fltk_gl_SHARED.lib" kann nicht geÃ¶ffnet werden.

===============================================
Last successful compilation and installation:
===============================================

 Verzeichnis von C:\Users\Public\bin

10.06.2020  19:19           542.720 libfltk_gl_SHARED.dll

 Verzeichnis von C:\Users\Public\lib

31.03.2022  20:44           440.812 fltk_gl.lib
10.06.2020  19:19           235.358 fltk_gl_SHARED.lib

===============================================
I had to make a change in CMake/options.cmake :
===============================================

#set (HAVE_GL LIB_GL OR LIB_MesaGL)

===============================================
Only now compilation and linking were successful:
===============================================

 Verzeichnis von C:\Users\Public\bin

15.04.2022  00:54             9.977 fltk-config
15.04.2022  01:00         1.179.136 libfltk.dll
10.06.2020  19:16           495.616 libfltk_forms_SHARED.dll
10.06.2020  19:19           542.720 libfltk_gl_SHARED.dll
10.06.2020  19:18           597.504 libfltk_images_SHARED.dll
10.06.2020  19:18         1.021.952 libfltk_SHARED.dll

 Verzeichnis von C:\Users\Public\lib

15.04.2022  00:57         4.776.690 fltk.lib
15.04.2022  00:57            88.712 fltk_forms.lib
10.06.2020  19:16           218.174 fltk_forms_SHARED.lib
31.03.2022  20:44           440.812 fltk_gl.lib
10.06.2020  19:19           235.358 fltk_gl_SHARED.lib
15.04.2022  00:57           621.460 fltk_images.lib
10.06.2020  19:18           212.626 fltk_images_SHARED.lib
10.06.2020  19:18         1.243.570 fltk_SHARED.lib

winfried
```

**Comment by Albrecht-S** [comment]
Thanks for the report. I know that there were recent changes in the CMake build files, particularly related to unittest-shared.exe but ... this is still work in progress and I see that such a build with MSVC (Visual Studio and/or NMake) and shared libraries ON doesn't work. I'm working on it. However, I just built w/o shared libs successfully. So for now I suggest to build w/o shared libs.

I'll let you know when building with shared libs and MSVC/NMake/Visual Studio works.

**Comment by szukw000** [comment]
@Albrecht-S , @ManoloFLTK ,
I have found an OLD BUG in a NEW LIBRARY.
```
[ 99%] Linking CXX executable ..\bin\test\unittests-shared.exe
LINK: command "C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1431~1.311\bin\Hostx86\x64\link.exe /nologo @CMakeFiles\unittests-shared.dir\objects1.rsp /out:..\bin\test\unittests-shared.exe /implib:..\bin\test\unittests-shared.lib /pdb:C:\Users\szukw000\LIB\FLTK\V1.4\fltk-master\BUILD\bin\test\unittests-shared.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:windows fltk_gl_SHARED.lib ..\lib\libfltk.lib opengl32.lib glu32.lib gdiplus.lib comctl32.lib ws2_32.lib gdiplus.lib C:\Users\Public\lib\jpeg.lib C:\Users\Public\lib\png.lib C:\Users\Public\lib\z.lib opengl32.lib glu32.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:..\bin\test\unittests-shared.exe.manifest" failed (exit code 1181) with the following output:
LINK : fatal error LNK1181: Eingabedatei "fltk_gl_SHARED.lib" kann nicht geÃ¶ffnet werden.
```
winfried

**Comment by Albrecht-S** [comment]
As I wrote already, this is work in progress (currently stalled), and the (full) CMake build support is new in FLTK 1.4. In the "old" library (FLTK 1.3.x) CMake support existed but was clearly said to be experimental. The Visual Studio build files bundled with FLTK 1.3 should work, although the shared libraries may be different, i.e. only one shared library rather than fltk, fltk_gl, fltk_images etc. (I'm not sure and I can't look it up right now).

**Comment by Albrecht-S** [comment]
This issue has been fixed in FLTK 1.4 (Git) a long time ago. To be sure I tested it with a commandline similar to the one given by @szukw000 (using CMake + nmake), and it worked as expected.

@szukw000 If there is still something wrong with your build, please open a new issue or ask in GitHub Discussions (Q&A).
